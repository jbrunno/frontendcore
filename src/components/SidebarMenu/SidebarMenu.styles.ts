import { styled } from '@mui/material';
import List from '../List/List';

export const SidebarList = styled(List)`
  display: grid;
  row-gap: ${({ theme }) => theme.fn.spacing(8)};
  grid-auto-rows: min-content;
  overflow-y: auto;
  overflow-x: hidden;
`;
