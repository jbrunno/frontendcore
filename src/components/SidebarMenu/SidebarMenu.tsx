import React, { ReactNode } from 'react';
import {
  SidebarMenuItem,
  SidebarMenuItemType,
} from '../SidebarMenuItem/SidebarMenuItem';
import { SidebarList } from './SidebarMenu.styles';

export type SidebarMenuProps = {
  isExpanded?: boolean;
  items?: SidebarMenuItemType[];
  children?:
    | ((props: Omit<SidebarMenuProps, 'children'>) => ReactNode)
    | ReactNode;
};

export function SidebarMenu({
  isExpanded = false,
  items,
  children,
}: SidebarMenuProps) {
  const renderChildren =
    typeof children === 'function' ? children({ isExpanded, items }) : children;

  return (
    <SidebarList>
      {!children && items
        ? items.map((item) => (
            <SidebarMenuItem
              key={item.label}
              item={item}
              isExpanded={isExpanded}
            />
          ))
        : renderChildren}
    </SidebarList>
  );
}

export default SidebarMenu;
