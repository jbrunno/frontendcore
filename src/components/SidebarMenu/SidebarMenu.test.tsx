import React from 'react';
import { render } from '@test-utils/component-testing';
import { Dashboard as DashboardIcon } from '@mui/icons-material';
import { SidebarMenu } from './SidebarMenu';

const items = [
  {
    icon: <DashboardIcon />,
    handle: () => null,
    label: 'Dashboard',
    match: true,
  },
];

test('renders the SidebarMenu', () => {
  const { container } = render(
    <SidebarMenu items={items} isExpanded={false} />,
  );
  expect(container).toMatchSnapshot();
});
