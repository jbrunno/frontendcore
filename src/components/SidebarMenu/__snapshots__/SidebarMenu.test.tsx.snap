// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SidebarMenu 1`] = `
.emotion-1 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  display: grid;
  row-gap: 0.5rem;
  grid-auto-rows: min-content;
  overflow-y: auto;
  overflow-x: hidden;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-width: 0;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding-left: 16px;
  padding-right: 16px;
  display: grid;
  grid-template-columns: auto;
  -webkit-column-gap: 0.5rem;
  column-gap: 0.5rem;
  border-radius: 4px;
  background-color: 08;
  overflow: hidden;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0000000A;
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-selected {
  background-color: rgba(74, 49, 255, 0.08);
}

.emotion-3.Mui-selected.Mui-focusVisible {
  background-color: rgba(74, 49, 255, 0.2);
}

.emotion-3.Mui-selected:hover {
  background-color: rgba(74, 49, 255, 0.12);
}

@media (hover: none) {
  .emotion-3.Mui-selected:hover {
    background-color: rgba(74, 49, 255, 0.08);
  }
}

.emotion-3.Mui-focusVisible {
  background-color: #0000001F;
}

.emotion-3.Mui-disabled {
  opacity: 0.38;
}

.emotion-3 .emotion-4 {
  color: #4A31FF;
}

.emotion-3 .e1dapwgc1 {
  color: #4A31FF;
}

.emotion-3:hover {
  background-color: #4A31FF08;
}

.emotion-3::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 1;
  top: 0;
  left: 0;
  -webkit-transform: scaleX(0) scaleY(50%);
  -moz-transform: scaleX(0) scaleY(50%);
  -ms-transform: scaleX(0) scaleY(50%);
  transform: scaleX(0) scaleY(50%);
  border-radius: 50%;
  background: radial-gradient(
          circle at center,
          #4A31FF12 25%,
          #4A31FF12 100%
        );
}

.emotion-3:focus::after {
  -webkit-transform: scaleX(110%) scaleY(300%);
  -moz-transform: scaleX(110%) scaleY(300%);
  -ms-transform: scaleX(110%) scaleY(300%);
  transform: scaleX(110%) scaleY(300%);
  opacity: 0;
  -webkit-transition: 0.4s ease-out,opacity 0.3s ease 0.4s;
  transition: 0.4s ease-out,opacity 0.3s ease 0.4s;
}

.emotion-3.Mui-disabled {
  opacity: 1;
}

.emotion-5 {
  min-width: 56px;
  color: #0000008A;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  min-width: -webkit-min-content;
  min-width: -moz-min-content;
  min-width: min-content;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<div>
  <ul
    class="MuiList-root MuiList-padding emotion-0 emotion-1"
  >
    <div
      aria-disabled="true"
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-disabled Mui-disabled emotion-2 emotion-3"
      role="button"
      tabindex="-1"
    >
      <div
        class="MuiListItemIcon-root emotion-4 emotion-5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
          data-testid="DashboardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
          />
        </svg>
      </div>
    </div>
  </ul>
</div>
`;
